dnf/base.py

```diff
@@ -838,6 +853,9 @@ def resolve(self, allow_erasing=False):
 
         goal.add_protected(self.sack.query().filterm(
             name=self.conf.protected_packages))
+
+        self._set_excludes_from_weak_to_goal()
```

Comment:
Could the new function in the context part of libdnf be used instead of duplicating it in dnf?

Reply:
It is good suggestion but it will be hard but not impossible, because the configuration class in python is quite difficult to pass to c code (The configuration class (C++) is exported by SWIG, then repacked by python in Class. Goal class (C++) is available in Python be C-Python). In DNF5 it must be not duplicated.